estimagic/optimization/optimize.py

```diff
@@ -217,7 +223,12 @@ def _single_minimize(
     """
     simplefilter(action="ignore", category=pd.errors.PerformanceWarning)
     params = _process_params(params)
-    fitness_eval = criterion(params, **criterion_kwargs)
+
+    fitness_factor = -1 if criterion_kwargs.get("_maximization", False) else 1
+    if "_maximization" in criterion_kwargs:
+        del criterion_kwargs["_maximization"]
```

Comment:
Why is ``criterion_kwargs["_maximization"]`` treated differently between https://github.com/OpenSourceEconomics/estimagic/pull/94/files#diff-4fc3d0959b0a8ca5740e893de422065bR75-R80 and the lines here? Can you handle them similary?

Reply:
I am not sure I understand what you want.

Would you prefer that I define a `fitness_factor` key with value 1 or -1 already in lines 75-80 instead of having `_maximization` as key and a boolean value?

Reply:
I should have read the code more carefully. The line with the fitness factor is great and handles both maximization and minimization problems.

Why do remove the key in the next lines? If it is necessary, use ``pop`` with default instead of ``get``: ``fitness_factor = -1 if criterion_kwargs.pop("_maximization", False) else 1``.

Reply:
Adding the maximization info to `criterion_kwargs` means I must remove it before they are passed on to `criterion` where the additional entry might cause problems. 

I guess the more general question is why pass `_maximization` as part of `criterion_kwargs` along and there is no good reason to do that anymore. Originally I tried to avoid changing the signature of `create_internal_criterion` but then decided it was worth to add `fitness_factor`.

Given that, I think it makes more sense to move `_maximization` to `general_options` where it does not have to me removed. Did that in bd369d47b0a8275b805a7b056b84af1d659cd512.

Reply:
Very good.